src/d/actor/d_a_obj_hami4.cpp

```diff
@@ -4,67 +4,192 @@
 //
 
 #include "d/actor/d_a_obj_hami4.h"
+#include "JSystem/J3DGraphAnimator/J3DModel.h"
+#include "d/d_com_inf_game.h"
 #include "d/d_procname.h"
 #include "d/d_priority.h"
 
 /* 00000078-00000098       .text CheckCreateHeap__FP10fopAc_ac_c */
-static BOOL CheckCreateHeap(fopAc_ac_c*) {
-    /* Nonmatching */
+static BOOL CheckCreateHeap(fopAc_ac_c* actor) {
+    return ((daObjHami4_c*)actor)->CreateHeap();
 }
 
 /* 00000098-00000374       .text CreateHeap__12daObjHami4_cFv */
-void daObjHami4_c::CreateHeap() {
-    /* Nonmatching */
+BOOL daObjHami4_c::CreateHeap() {
+    J3DModelData * modelData = (J3DModelData *)dComIfG_getObjectRes("Hami4", 0x4);
+    JUT_ASSERT(0x5f, modelData != NULL);
+
+    if (fopAcM_isSwitch(this, prm_get_swSave())) {
+        this->field_0x378 = 3;
+        this->field_0x37C = 1500.0;
```

Comment:
Missing `f` suffix for singles (also applies to all the other singles this in PR)
```suggestion
        this->field_0x37C = 1500.0f;
```